pipeline.py

```diff
@@ -72,31 +97,120 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            tokenized_text = self._process(article.get_raw_text())
+
+            cleaned = []
+            single_tagged = []
+            multiple_tagged = []
+
+            for token in tokenized_text:
+                cleaned.append(token.get_cleaned())
+                single_tagged.append(token.get_single_tagged())
+                multiple_tagged.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        pattern = re.compile(r'[А-Яа-яA-Za-z ёЁ]')
+
+        cleaned_text = raw_text
+
+        for symbol in raw_text:
+            if not pattern.match(symbol):
+                cleaned_text = raw_text.replace(symbol, '')
+
+        tokens = []
+
+        result = Mystem().analyze(cleaned_text)
+        analyzer = pymorphy2.MorphAnalyzer()
+
+        for single_word in result:
+
+            if 'analysis' not in single_word or not single_word['analysis']:
+                continue
+
+            token = MorphologicalToken(single_word['text'])
+            tokens.append(token)
+
+            token.normalized_form = single_word['analysis'][0]['lex']
+            token.tags_mystem = single_word['analysis'][0]['gr']
+            token.tags_pymorphy = analyzer.parse(single_word['text'])[0].tag
+
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+
+    if not path.exists():
+        raise FileNotFoundError
+
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    if not list(path.glob('**/*')):
+        raise EmptyDirectoryError
+
+    needed_file_names = ['_raw', '_meta']
+
+    files = {
+        ".json": [],
+        ".pdf": [],
+        ".txt": []
+    }
+
+    pattern = re.compile(r'\d+')
+
+    for file in list(path.glob('*')):
+
+        # check txt files
+
+        # add in dict file id with relevant filename extension
+        for file_name_type in needed_file_names:
+            if file_name_type in file.stem:
+                files.get(file.suffix).append(int(pattern.match(file.stem).group()))
```

Comment:
who guaranteed that all files will have a number in the beginning?

Reply:
We expect to process the files given by `scrapper` via `Article.save_raw` method, which saves text files with their ids in the beginning

Reply:
Add a check: if there is no number in the beginning, we take the index of the sorted list of files.